lib/src/exec-graph.ts

```diff
@@ -1,47 +1,43 @@
 import { promises as fs } from 'node:fs';
-
+import {
+  DefaultLogger,
+  Engine,
+  Logger,
+  ManualLifecycleEventEmitter,
+  readGraphFromJSON,
+  registerCoreProfile,
+  registerSceneProfile,
+  Registry,
+  traceToLogger,
+  validateGraph,
+  validateRegistry,
+  writeGraphToJSON
+} from './';
 import { program } from 'commander';
 import glob from 'glob';
-
-import { Logger } from '../../lib/Diagnostics/Logger.js';
-import { Engine } from '../../lib/Execution/Engine.js';
-import { traceToLogger } from '../../lib/Execution/traceToLogger.js';
-import { readGraphFromJSON } from '../../lib/Graphs/IO/readGraphFromJSON.js';
-import { writeGraphToJSON } from '../../lib/Graphs/IO/writeGraphToJSON.js';
-import { validateGraph } from '../../lib/Graphs/Validation/validateGraph.js';
-import { DefaultLogger, ManualLifecycleEventEmitter } from '../../lib/index.js';
-import { parseSafeFloat } from '../../lib/parseFloats.js';
-import { registerCoreProfile } from '../../lib/Profiles/Core/registerCoreProfile.js';
-import { registerSceneProfile } from '../../lib/Profiles/Scene/registerSceneProfile.js';
-import { Registry } from '../../lib/Registry.js';
-import { validateRegistry } from '../../lib/validateRegistry.js';
-
-async function main() {
-  //Logger.onVerbose.clear();
-
-  program
-    .name('exec-graph')
-    .argument('<filename>', 'path to the behavior-graph json to execute')
-    .option('-t, --trace', `trace node execution`)
-    .option('-p, --profile', `profile execution time`)
-    .option('-d, --dryRun', `do not run graph`)
-    .option(
-      '-u, --upgrade',
-      `write json graph back to read location, upgrading format`
-    )
-    .option('-i, --iterations <iterations>', 'number of tick iterations', '5');
-
-  program.parse(process.argv);
-  const programOptions = program.opts();
-
+import { parseSafeFloat } from './parseFloats';
+
+type ProgramOptions = {
+  upgrade?: boolean,
+  trace?: boolean,
+  dryRun?: boolean,
+  profile?: boolean,
+  iterations: string 
+};
+
+function execGraph({
```

Comment:
just a minor refactoring - in case we want to just export this function and call it directly from the outside world